src/h5web/visualizations/shared/VisCanvas.tsx

```diff
@@ -66,14 +76,19 @@ function VisCanvas(props: Props): JSX.Element {
             gl={{ preserveDrawingBuffer: true }} // for screenshot feature
           >
             <ambientLight />
-            {axisDomains && (
-              <AxisSystem
-                axisDomains={axisDomains}
-                axisOffsets={axisOffsets}
-                showGrid={showGrid}
-              />
-            )}
-            {children}
+            <AxisSystemProvider
+              showGrid={showGrid}
+              hasXLogScale={hasXLogScale}
+              hasYLogScale={hasYLogScale}
```

Comment:
We could pass the domains as well so the `useAbscissa` and `useOrdinate` hooks can retrieve them themselves. This might be useful for the tooltip and other features as well.